### PR TITLE
Fix lost aspnet.createComponent calls, part II (fixes T886572)

### PR DESCRIPTION
**Pull request by AlekseyMartynov**
I considered this back in September. Check 'Keep the handler in queue until the element is found' in #9540.

Now, thanks to ticket https://devexpress.com/issue=T886572, we have a test case that confirms the actual need.

### CODE REVIEW

**Review by nightskylark — COMMENTED** [review]
LGTM

**Review by DokaRus — APPROVED** [review]

File: js/aspnet.js [hunk 1]
```diff
@@ -143,17 +143,29 @@
...
+        var leftover = [ ];
+
         pendingCreateComponentRoutines = [ ];
-        snapshot.forEach(function(func) { func(); });
+        snapshot.forEach(function(func) {
+            if(!func()) {
+                leftover.push(func);
+            }
+        });
+        pendingCreateComponentRoutines = pendingCreateComponentRoutines.concat(leftover);
```
Comment (San4es) [inline js/aspnet.js:168]: It looks like `pendingCreateComponentRoutines` is an empty array here. Can we just set `leftover` to it?
```suggestion
        pendingCreateComponentRoutines = leftover;
```

File: js/aspnet.js [hunk 2]
```diff
@@ -143,17 +143,29 @@
...
 
     templateRendered.add(function() {
         var snapshot = pendingCreateComponentRoutines.slice();
+        var leftover = [ ];
+
         pendingCreateComponentRoutines = [ ];
-        snapshot.forEach(function(func) { func(); });
+        snapshot.forEach(function(func) {
+            if(!func()) {
+                leftover.push(func);
```
Comment (San4es) [inline js/aspnet.js:165]: Can we push items to `pendingCreateComponentRoutines` directly?
```suggestion
                pendingCreateComponentRoutines.push(func);
```

File: js/aspnet.js (hunk 1 above)
Comment (San4es) [reply js/aspnet.js:168]: We found that `pendingCreateComponentRoutines` can be a not empty array when it is called recursively (for instance, during nested components rendering).

**Comment by AlekseyMartynov** [comment]
@San4es 

- `snapshot` and `leftover` are function-local variables
- `pendingCreateComponentRoutines` is module-local
- `pendingCreateComponentRoutines` *might be* modified in a recursive `templateRendered` call. 
- `snapshot` isolates the inbound state for correct iteration
- `leftover` isolates outbound changes

You suggestions may be valid, however I didn't succeed in authoring a relevant test case.

**Review by San4es — APPROVED** [review]
LGTM

**Pull request merged**